backend/src/api/email/email.handlers.ts

```diff
@@ -1,19 +1,51 @@
 import { Request, Response } from "express-serve-static-core";
-import { PostmarkInboundEmailBody } from "./email.types";
+import { MailgunInboundEmailBody } from "./email.types";
+import {
+  parseUnverifiedVenmoEmail,
+  updateOrderPaymentStatus,
+} from "./email.services";
+import { isValidVenmoEmail } from "../../utils/email";
+import { z } from "zod";
 
-// TODO
 export const parseEmailHandler = async (
-  req: Request<{}, any, PostmarkInboundEmailBody, {}>,
+  req: Request<{}, any, z.infer<typeof MailgunInboundEmailBody>, {}>,
   res: Response
 ) => {
-  // https://postmarkapp.com/developer/user-guide/inbound/parse-an-email
+  try {
+    const bodyHtml = req.body["body-html"];
+    const bodyPlain = req.body["Body-plain"];
+    const { from, subject } = req.body;
 
-  // parse email HtmlBody
-  req.body.HtmlBody;
+    // Verify email is from Venmo
+    const isFromVenmo = isValidVenmoEmail(from);
+    if (!isFromVenmo) {
+      res.status(400).json({
+        message: "Email not from verified Venmo address",
+      });
+      return;
+    }
 
-  // verify email
+    // Use HTML body if available, otherwise use plain text
+    const emailContent = bodyHtml || bodyPlain;
 
-  // save email transaction to database
+    // Parse email to extract amount and order ID
+    const { parsedAmount, orderId } = parseUnverifiedVenmoEmail(emailContent);
```

Comment:
We need to incorporate some branching logic to decide whether we go to parseUnverifiedVenmoEmail or parseVerifiedVenmoEmail

Reply:
We should also add a try/catch because this function might fail